### PR TITLE
OpenFOAM-foss-2019a

### PR DESCRIPTION
**Pull request by furstj**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by furstj** [comment]
OpenFOAM-6 and v1812 fails on HP Superdome NUMA systems. The bug is probably related to OpenMPI prior to 3.1.2 (see https://raw.githubusercontent.com/open-mpi/ompi/v4.0.x/NEWS mentioning race condition in the section mentionin 3.1.2 version).

Together with toolchain update the OpenFOAM was compiled with new ParaView supporting FFmpeg (for making animations) and using Qt5 5.9 LTS version.

File: easybuild/easyconfigs/f/FFmpeg/FFmpeg-4.1-foss-2019a.eb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+easyblock = 'ConfigureMake'
+
+name = 'FFmpeg'
```
[the remark this answers is not in the document]
Comment (furstj) [reply easybuild/easyconfigs/f/FFmpeg/FFmpeg-4.1-foss-2019a.eb]: I will delete FFmpeg in next commit.

File: easybuild/easyconfigs/l/LLVM/LLVM-6.0.0-GCCcore-8.2.0.eb [hunk 2]
```diff
@@ -0,0 +1,45 @@
+easyblock = 'CMakeMake'
+
+name = 'LLVM'
```
[the remark this answers is not in the document]
Comment (furstj) [reply easybuild/easyconfigs/l/LLVM/LLVM-6.0.0-GCCcore-8.2.0.eb]: Paraview 5.6.0 does not compile with Mesa 19 from easybuild. Therefore I have to compile it with Mesa 18. Unfortunately the Mesa 18 requires LLVM 6. Therefore LLVM6 is still required.

File: easybuild/easyconfigs/l/libGLU/libGLU-9.0.0-GCCcore-8.2.0-Mesa-18.3.6.eb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+easyblock = 'ConfigureMake'
+
+name = 'libGLU'
```
[the remark this answers is not in the document]
Comment (furstj) [reply easybuild/easyconfigs/l/libGLU/libGLU-9.0.0-GCCcore-8.2.0-Mesa-18.3.6.eb]: Yes, but the GLU from easybuild develop uses Mesa 19 and therefore breaks ParaView 5.6.0 (see my notice on LLVM 6 above)

File: easybuild/easyconfigs/m/MPFR/MPFR-4.0.1-GCCcore-8.2.0.eb [hunk 4]
```diff
@@ -0,0 +1,34 @@
+easyblock = 'ConfigureMake'
+
+name = 'MPFR'
+version = '4.0.1'
```
[the remark this answers is not in the document]
Comment (furstj) [reply easybuild/easyconfigs/m/MPFR/MPFR-4.0.1-GCCcore-8.2.0.eb]: I will delete  my MPFR in next commit and replace it with MPFR 4.0.2

File: easybuild/easyconfigs/m/Mesa/Mesa-18.3.6-GCCcore-8.2.0.eb [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+# not hardware rendering. This means you want at least SSE4.2. We build:
+# - llvmpipe: the high-performance Gallium LLVM driver
+# - swr: Intel's OpenSWR
+# it will try to use the llvmpipe by default. It you want swr, do:
+# GALLIUM_DRIVER=swr
+
+easyblock = 'ConfigureMake'
+
+name = 'Mesa'
+version = '18.3.6'
```
[the remark this answers is not in the document]
Comment (furstj) [reply easybuild/easyconfigs/m/Mesa/Mesa-18.3.6-GCCcore-8.2.0.eb]: Yes, but ParaView 5.6.0 does not compile with Mesa 19 from easybuild (there are some problems with linking, may be due to missing *.la files in Mesa 19). Therefore Mesa 18 is still needed.

File: easybuild/easyconfigs/q/Qt5/Qt5-5.9.8-foss-2019a.eb [hunk 6]
```diff
@@ -0,0 +1,30 @@
+easyblock = 'EB_Qt'
+
+name = 'Qt5'
+version = '5.9.8'
```
[the remark this answers is not in the document]
Comment (furstj) [reply easybuild/easyconfigs/q/Qt5/Qt5-5.9.8-foss-2019a.eb]: Qt5 5.12.3 is linked against libGLU which uses Mesa 19 (see my comments above).

File: easybuild/easyconfigs/m/MPFR/MPFR-4.0.1-GCCcore-8.2.0.eb (hunk 4 above)
[the remark this answers is not in the document]
Comment (furstj) [reply easybuild/easyconfigs/m/MPFR/MPFR-4.0.1-GCCcore-8.2.0.eb]: Finally I chose to move MPFR 4.0.2 to a standalone PR (see PR #8240).

**Comment by Micket** [comment]
I'll write a comment, as it pertains to several configs: The duplicate Mesa is not something to shrug off. IT creates a new branch of several duplicate, incompatible, packages within the same toolchain.

I'd very much like to see that linking issue with Mesa 19 resolved so that we don't create so many problems for ourselves in the future.

Now, if (big if) we truly have to create a duplicate Mesa, then, Qt5 should still go into GCCcore (and, probably, no need to pick an old Qt version as well?). Every package that that even indirectly depends on this old Mesa needs to reflect this by a suffix. Still, I strongly suggest we take a look at why there was an issue building to begin with.

(also note on my review: I didn't actually get so far as to review the OpenFOAM packages themselves)

**Comment by furstj** [comment]
I agree that duplicate Mesa is not at all optimal solution. I will try to resolve linking problems with Mesa 19 & ParaView (or VTK) at first.

**Comment by Micket** [comment]
@furstj If you could upload a test report for #8225 I would appreciate it
```
eb --from-pr 8225 --upload-test-report -r
```

I was actually trying to build Paraview against this VTK library myself. Not sure if the problem is in VTK, or ParaView, but I got stuck at the configure step, as ParaView couldn't find the vtkCommonCore.cmake files (though, they seem to have been installed correctly).
This is because I tried to tweak a few things in VTK, avoiding internal third party components (that we have compiled anyway, like mpi4py). It would seem prudent for a ParaView build to make use of this, so, `-DUSE_EXTERNAL_VTK=on`)

**Comment by furstj** [comment]
I will do it. But the compilation takes long time...

**Comment by furstj** [comment]
Thanks for #8225, I will try ParaView with VTK.

**Pull request closed without merging**